syntax = "proto3";
package open_app_desc;
enum BODY_TYPE
{
	BODY_BEGIN = 0;
    BODY_REQUEST = 1;
    BODY_RESPONSE = 2;
};

message CmdBase
{
	int64 seq_id = 1; 
	int64 cmd_id = 2;
	int64 type = 3; 
	bytes body = 4;      
};

message CmdBaseResp
{
	int64 ret = 1;      
	bytes err_msg= 2;     
	bytes resp = 3;      
};

message L5cfg
{
	bytes modId = 1;
	bytes cmdId = 2;
	bytes defaultIp = 3;
	int64 defaultPort = 4;
	bytes globalDefaultIp = 5;
	int64 globalDefaultPort = 6;
}

message Flux{
	uint32 flowswitch = 1; //已加 //流控开发 值不存在或者1 流控打开， 0 表示流控关闭
	uint32 queryrate = 2;//已加 //活动查询 流量限制百分比 0-100%，0%：该实例查询完全关闭 100%：该实例查询不做任何限流(缺省是100%，没有该值也表示不做任何限流)
	uint32 attendrate = 3;//未加 //活动参与 流量限制百分比 0-100%，0%：该实例查询完全关闭 100%：该实例查询不做任何限流(缺省是100%，没有该值也表示不做任何限流)
	uint32 queryid = 4; //已加 //tnm2 活动查询上报ID 管理端分配
	uint32 attendid = 5; //已加 //tnm2 活动参与上报ID 管理端分配
	uint32 totalflowid = 6; //已加 //tnm2 活动总请求量上报ID 管理端分配
	uint32 queryquota = 7; //待加 //活动查询每秒最大请求量(单位:个) 缺省:0个
	uint32 attendquota = 8; //未加 //活动参与每秒最大请求量(单位:个) 缺省:0个
	bytes  referkey = 9; //待加 //活动配置参考key(如：其它活动ID)
	uint32 winlen = 10; //已加 //回溯时间，1-30秒 缺省6秒
	uint32 errnum = 11; //已加 //流控错误触发数 >=10次 缺省50次
	uint32 errnumrate = 12; //已加 //流控错误触发比例 1-99% 缺省 10%
	uint32 cpurate = 13; //已加 //CPU负载触发百分比   60-99% 缺省 90%
	uint32 cpuinterval = 14; //已加 //CPU负载获取间隔时间(second) 1-60秒 缺省 3秒
	uint32 memrate = 15; //未加 //Memory负载触发百分比 60-99% 缺省 90%
	uint32 queryquovaryid = 16; //未加 //活动查询最大请求量变化上报ID
	uint32 attendquovaryid = 17; //未加 //活动参与最大请求量变化上报ID
	L5cfg  normalenv = 18;//待加 未加 
	L5cfg	testenv = 19;//待加 未加
	uint32 type = 20;// 1:effective 2:accurate
}

// quota
message ReportItem{
	int64 timekey = 1;
	int64 reportnum = 2;
	int64 refuse = 3;
	int64 setrefuse = 4;
}

message QuotaReq
{
	bytes key = 1; // quote key: act, instanceid
	bytes referkey = 2;
	repeated ReportItem reportitem = 3;
	int32 localquota = 4; // 
	int32 id = 5; // process group id
	bytes serial = 6; //
	int32 totalquota = 7; // key total quota
	bytes openid = 8;
	int64 uin = 9;
	int64 nowsum = 10; // percent seconds
	int64 workernum = 11;
	bytes setkey =12;
	bytes business = 13; //
	bytes setenv = 14; // 
}

message QuotaRsp
{
	bytes key = 1;
	int64 localq = 2; // totalq*(self)/total
	int64 totalq = 3;
	int32 partnernum = 4;
	bytes serial = 5; //
	int32 ipnum = 6;
	int32 alloctype = 7;// 0:undo 1:average 2:dynamic 3:heart beat reset 4:history总量不够
	int32 allocpartnum = 8;//
	int64 alloctime = 9; //
	int64 self = 10; //
	int64 total = 11; //
	int64 nowsum = 12; //
	int64 totalworkernum = 13; //
	int32 setquota = 14; //
	int32 setsum = 15; //
	int32 procquota = 16; //
}

// machine config
message SetItem
{
	bytes setkey = 1;
	int32 setquota = 2;
	repeated bytes iplist = 3;
	int32 feaid = 4;
	int32 refuseid = 5;
	int32 setrefuseid = 6;
	int32 realid = 7;
}

message SetMacInfo
{
	repeated SetItem itemlist = 1;//
}

message SetFileInfo
{
	bytes fpath = 1; 
	repeated bytes fname = 2;
}

// to second aquota
message ProcReport
{
	bytes procid = 1; //
	repeated ReportItem proclist = 2;
	bytes setkey = 3;
}

message SetQuotaReq
{
	bytes setkey = 1;
	repeated ReportItem itemlist = 2;
	int32 workernum = 3;
	int32 id = 4; // spp group id
	repeated ProcReport procitem = 5;
}

message ProcQ
{
	bytes procid =1;
	int32 procquota = 2; //
}

message SetQuotaRsp
{
	bytes setkey = 1;
	int32 setsum = 2; //
	repeated ProcQ procq = 3; //
}

// active/ip/set report
message RealReport{
	int32 keytype = 1; // 1:key 2: machine 3: SET 
	bytes keyname = 2; 
	int64 timekey = 3;
	bytes setname = 4;
	int32 total = 5;
	int32 refuse = 6;
	int32 setrefuse = 7;
	int32 real = 8;
	int32 quota = 9;
}

message QReport
{
	bytes actid = 1; // 1:
	bytes ip = 2;
	int32 procid = 3;
	int64 timekey = 4;
	bytes setname = 5;
	int32 actquota = 6;
	int32 setquota = 7;
	int32 total = 8;	
	int32 refuse = 9;
	int32 setrefuse = 10;
	int32 real = 11;
}

